caravel/assets/javascripts/SqlLab/components/TableElement.jsx

```diff
@@ -36,7 +38,16 @@ class TableElement extends React.Component {
     if (this.props.table.schema) {
       tableName = this.props.table.schema + '.' + tableName;
     }
-    return `SELECT ${cols}\nFROM ${tableName}`;
+    let sql;
+    if (useStar) {
+      sql = `SELECT * FROM ${tableName}`;
+    } else {
+      sql = `SELECT ${cols}\nFROM ${tableName}`;
+    }
+    if (limit > 0) {
+      sql += `\nLIMIT ${limit}`;
+    }
+    return sql;
```

Comment:
sucks to have the logic on both sides, but it isn't great to call the server when all the necessary info is already in the state...